### ISSUE TITLE
1.0.2: some of the errors are not suppressed

### ISSUE THREAD

**Issue by skshetry** [reverbc/pylint-pytest#13]
The new 1.0.2 release is not suppressing errors.
```console
$ pylint tests

ests/unit/test_updater.py:29:22: W0621: Redefining name 'tmp_global_dir' from outer scope (line 14) (redefined-outer-name)
tests/unit/test_updater.py:29:22: W0613: Unused argument 'tmp_global_dir' (unused-argument)
tests/unit/test_updater.py:39:25: W0621: Redefining name 'updater' from outer scope (line 29) (redefined-outer-name)
tests/unit/test_updater.py:64:25: W0621: Redefining name 'updater' from outer scope (line 29) (redefined-outer-name)
tests/unit/test_updater.py:73:57: W0621: Redefining name 'updater' from outer scope (line 29) (redefined-outer-name)
tests/unit/test_updater.py:88:23: W0621: Redefining name 'mock_tty' from outer scope (line 34) (redefined-outer-name)
tests/unit/test_updater.py:88:33: W0621: Redefining name 'updater' from outer scope (line 29) (redefined-outer-name)
tests/unit/test_updater.py:88:23: W0613: Unused argument 'mock_tty' (unused-argument)
tests/unit/test_updater.py:103:34: W0621: Redefining name 'mock_tty' from outer scope (line 34) (redefined-outer-name)
tests/unit/test_updater.py:103:44: W0621: Redefining name 'updater' from outer scope (line 29) (redefined-outer-name)
tests/unit/test_updater.py:103:34: W0613: Unused argument 'mock_tty' (unused-argument)
tests/unit/test_updater.py:121:4: W0621: Redefining name 'mock_tty' from outer scope (line 34) (redefined-outer-name)
tests/unit/test_updater.py:121:14: W0621: Redefining name 'updater' from outer scope (line 29) (redefined-outer-name)
tests/unit/test_updater.py:121:4: W0613: Unused argument 'mock_tty' (unused-argument)
tests/unit/test_updater.py:135:27: W0621: Redefining name 'updater' from outer scope (line 29) (redefined-outer-name)
``` 

Looking at the recent changes, it seems to be that the following lines are responsible:

https://github.com/reverbc/pylint-pytest/blob/be2a4080c9d38f60451c36596b011e65f564f911/pylint_pytest/checkers/fixture.py#L91-L92

It works without these 2 lines, so I think some modules get removed that are required later for the `checker`. I am leaving an issue here as someone here might have a good idea around it. Apologies for not providing a script, I'll try to come up with one.

**Comment by reverbc** [comment]
Thanks for reporting the issue @skshetry! Looks like it's the `inspect` module that is not happy about my decision to del imported modules at https://github.com/reverbc/pylint-pytest/blob/pylint-pytest-1.0.2/pylint_pytest/utils.py#L96

That's actually an interesting issue. I got the fixture details with https://github.com/reverbc/pylint-pytest/blob/pylint-pytest-1.0.2/pylint_pytest/checkers/fixture.py#L17 which gives me full `<class 'function'>` object of all the available fixtures. I rely on `inspect.getmodule` to find the python file where this function was defined, but in that method it'll just read from `sys.modules` if the function object has `__module__` attribute (which it has!) (https://github.com/python/cpython/blob/3.9/Lib/inspect.py#L731-L732). Of course it can't find anything because the module was already deleted so it returns `None` and thus the plugin cannot make sure the name it's suspecting to be an unused import is a fixture.

For this particular issue I think it's easy to fix by using `inspect.getabsfile` method directly (which is used by `inspect.getmodule` if there's no `__module__` attribute). However for the other `unused-argument` and `redefined-outer-name` errors I don't have a good guess on what might be the cause yet as I have no code that can reproduce the regression. Is it possible for you to generate some code snippet that can demonstrate the issue?

**Comment by reverbc** [comment]
Fixed with #15 & 1.0.3

**Comment by skshetry** [comment]
@reverbc, can confirm that it fixed the issue. Thanks a lot for creating this plugin. 🙏🏼
